alchemist/alchemist-cognitive-agents/src/main/kotlin/it/unibo/alchemist/model/implementations/actions/IterableExtension.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2010-2020, Danilo Pianini and contributors
+ * listed in the main project's alchemist/build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.model.implementations.actions
+
+import it.unibo.alchemist.model.implementations.positions.Euclidean2DPosition
+
+/**
+ * Takes the first [percentage] * size elements of the list.
+ */
+fun <T> List<T>.takePercentage(percentage: Double): List<T> = take((percentage * size).toInt())
+
+/**
+ * Converts an array of numbers representing [Euclidean2DPosition]s to an actual list of positions.
+ */
+fun Array<out Number>.toPositions(): List<Euclidean2DPosition> =
+    toList().chunked(2) { Euclidean2DPosition(it[0].toDouble(), it[1].toDouble()) }
```

Comment:
Woah, this is broken under many conditions. How frequently do you use it? If you use it in a single place, move it there.
Otherwise, you need to pass the environment as parameter, chunk by dimensions and use makePosition to build the correct position.

Reply:
It's used widely, fixed as indicated.